apis.md

```diff
@@ -9,3 +9,10 @@ The following OpenTracing APIs are currently available in the RFC stage:
 
 Please refer to the README files in the respective projects for examples of usage.
 
+### Implementation Notes
+
+OpenTracing APIs intend to express a similar experience to the user, while still being idiomatic to the language or framework. The following concepts apply to all variants.
+
+A **Span** is shared mutable state. It accumulates tags and logs, and can also create children. Finishing a span means capturing its state as a SpanRecord.
+
+A **SpanRecord** is an immutable representation of a completed span. It can be reported directly to an OpenTracing system, or encoded into a representation of an alternate system, such as HTrace or Zipkin.
```

Comment:
>
> that's what I've implemented in api-go as well, InMemoryRecorder - https://github.com/opentracing/api-go/blob/master/testutils/recorder.go#L13
>
> yeah that looks like same concept at first glance.

> api-python/examples/zipkin-like has a LoggingReporter, which could use
> some improvement on the output format as it just calls a toString on the
> span. An InMemoryReporter can be added easily.
>
ack

> So I think zipkin-like package satisfies your reqs, but it's a real
> implementation, not a fake, just depends on which reporter you use it with.
> I didn't give it a real reporter because that depends on the collector
> implementation - could be http post to query svc, or save to kafka. One we
> move it to openzipkin repo, we'd want to add some real reporter, or two.
>
I think mostly same page. I think the zipkin-like becomes zipkin real in
the repo as you mentioned, but I do think something that defines tests
shouldn't live downstream.

IOTW, at least in api-java I plan to use the same route as we do in zipkin
for span stores. Which is to define tests and a fake that live in the
authoritative repo. Downstream implementations must pass or explicitly skip
these tests. This simplifies answer to the question of what behavior should
be vs what one or another downstream do.